bin/jacodbg

```diff
@@ -0,0 +1,80 @@
+#!/bin/sh
+
+resolve()
+{
+    CMD="$1"
+    # test if CMD is a symlink or not
+    while [ -h "$CMD" ] ; do
+        link=`find "$CMD" -printf "%l\n"`
+        if expr "$link" : '/.*' > /dev/null; then
+            CMD="$link"
+        else
+            CMD=`dirname "$CMD"`"/$link"
+        fi
+    done
+    echo $CMD;
+}
+
+case $0 in
+    /*)
+        _SELF="$0"
+        ;;
+    *)
+        _SELF="`pwd`/$0"
+        ;;
+esac
+
+dbgport=5005
+pass=()
+pndx=0
+while [[ $# > 0 ]]
+do
+    case $1 in
+        -dbgport)
+            dbgport=$2
+            shift
+            ;;
+        *)
+            pass[$pndx]=${1}
+            pndx=`expr $pndx + 1`
+            ;;
+    esac
+    shift
+done
+
+_SELF=`resolve "$_SELF"`
+_BIN_DIR_RAW=`dirname "$_SELF"`
+_BIN_DIR_NORM=`cd "$_BIN_DIR_RAW" > /dev/null 2>&1; pwd`
+RESOLVED_JACORB_HOME=`dirname "$_BIN_DIR_NORM"`
+
+JACORB_HOME=${RESOLVED_JACORB_HOME}
+
+# Test if JRE_HOME or JAVA_HOME provide java-binary
+# if not fall back to path
+if [ -n "$JRE_HOME" ] && [ -x "$JRE_HOME/bin/java" ]
+then
+    RESOLVED_JAVA_CMD=$JRE_HOME/bin/java
+else
+    if [ -n "$JAVA_HOME" ] && [ -x "$JAVA_HOME/bin/java" ]
+    then
+        RESOLVED_JAVA_CMD=$JAVA_HOME/bin/java
+    else
+        RESOLVED_JAVA_CMD=java
+    fi
+fi
+
+JAVA_CMD=${RESOLVED_JAVA_CMD}
+
+# verbosity output
+#echo    "using JAVA_CMD   : ${JAVA_CMD}"
+#echo    "Using JacORB from: ${JACORB_HOME}"
+#echo -e "using CLASSPATH  :\n\t`echo $CLASSPATH | sed -e 's/:/\n\t/g'`"
+
+exec "$JAVA_CMD"                                                    \
+     -Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=${dbgport} \
+     -Djava.endorsed.dirs="${JACORB_HOME}/lib"                      \
```

Comment:
Because I'm old and wouldn't be able to remember -Xdebug.... So for me, creating a separate script made it clear at a glance what's going on with the least amount of extra business. Of course you are right that now there is a separate script that has to be managed. It's probably better move the -Xdebug setting line into the existing jaco and have a simple command line option to enable that. Then make jacodbg a wrapper that simply invokes jaco with the additional parameter(s). I still like the idea of two scripts, so that way in a year or two when I need to use it again, I can just list the bin directory to be reminded which script to use.